### ISSUE TITLE
Using toString () on SelenideElement returns NoSuchElementException on Android

### ISSUE THREAD

**Issue by dkorobtsov** [selenide/selenide#496]
Discovered during testing of: :https://github.com/codeborne/selenide/commit/ae307134cba149a90f64a427a80ad56badb28bba

Use case:

```
    public Boolean validateUiElementCondition(final SelenideElement element, Condition condition){
        Boolean mResult;
        if (element != null)  {

            log.info("Checking if '" + Helper.trimElementName(element.toString()) + "' is " + condition.toString() + '.');
            if ($(element).is(condition)) {
                log.info(Helper.trimElementName(element.toString()) + " is " + condition.toString());
                mResult = true;
            } else {
                log.error(Helper.trimElementName(element.toString()) + " is not " + condition.toString());
                mResult = false;
            }
        } else {
            mResult = null;
        }
        return mResult;
    }
```

Output:
[Invoker 2110312532] Invoking tests.ui_validation_tests.LoginUiTest.testLoginUiMeetsTheRequirements
2017-03-17 01:08:23.566 [main] INFO  LoginUiTest:43 - Starting UI element condition validation.
2017-03-17 01:08:23.626 [main] INFO  Validator:38 - Checking if 'NoSuchElementException: An element could not be located on the page using the given search parameters.' is visible.
2017-03-17 01:08:23.813 [main] INFO  Validator:40 - NoSuchElementException: An element could not be located on the page using the given search parameters. is visible

![zzzzzz](https://cloud.githubusercontent.com/assets/13999072/24022416/f9ade206-0aae-11e7-8eff-a26b5c432595.PNG)

It would be really nice to get as output of toString() element's name instead of exceptions.

For example:
SelenideElement button = $(By.id(APP_ID + "button_name"));
button.toString()   ==> APP_ID + "button_name"

**Comment by dkorobtsov** [comment]
Update.
I suppose those exceptions could be one of the reasons of serious performance drop reported earlier:
https://github.com/codeborne/selenide/issues/457

Just a guess. At least on Android it's a major issue.

Basically, I have made two versions of same test suite for testing purposes.
Environment: 
testng+selenide+appium, tests executed on real android device
~100 test cases, mostly dataprovider based.

Scenario 1: datatype of all Page Object fields - SelenideElement
Suite execution time: ~10 minutes

Scenario 2: datatype of all Page Object fields - By
Suite execution time: ~2 minutes

Check attached screenshot for proof:
![selenide_and_by_comparison](https://cloud.githubusercontent.com/assets/13999072/24063831/ff44e398-0b69-11e7-9c0f-4ad30314aa12.png)

**Comment by egorvas** [comment]
+1 sounds good if it real

**Comment by rmarinsky** [comment]
@dkorobtsov Can you share these tests on you account?

**Comment by dkorobtsov** [comment]
@rmarinsky 

Basically test is the one I've started topic with. I'm using it with along testng @dataprovider, test data is array of custom objects like:
{element1, condition1}
{element2, condition1}
{element3, condition1}
{element4, condition2}
{element5, condition3}
etc
This approach allows to validate all elements on a page in a matter of seconds using just one test method. (at least if we are working with "By"). 

I can share an example a bit later, need some time to prepare it :)

**Comment by asolntsev** [comment]
Fixed in Selenide 5.2.5
Now element.toString() looks like `<android.widget.Button name="2">2</android.widget.Button>`
